SolverManager/src/main/java/com/oops/solvermanager/SolverManagerController.java

```diff
@@ -0,0 +1,118 @@
+import com.oops.solvermanager.Requests.SolverBody;
+
+import io.fabric8.kubernetes.api.model.Config;
+import io.fabric8.kubernetes.api.model.batch.v1.Job;
+import io.fabric8.kubernetes.api.model.batch.v1.JobBuilder;
+import io.fabric8.kubernetes.client.ConfigBuilder;
+import io.fabric8.kubernetes.client.KubernetesClient;
+import io.fabric8.kubernetes.client.KubernetesClientBuilder;
+
+@SpringBootApplication
+@RestController
+public class SolverManagerController {
+    public static void main(String[] args) {
+
+        SpringApplication.run(SolverManagerController.class, args);
+    }
+
+    @PostMapping("/new")
+    public ResponseEntity<String> createJob(@RequestBody ProblemRequest newProblem) {
+        SolverBody test = newProblem.getSolversToUse()[0];
+        createSolverJobs(newProblem);
+        return ResponseEntity.status(HttpStatus.OK).body(test.getSolverName());
+    }
+
+    @PostMapping("/cancel/task/{problemID}")
+    public ResponseEntity<String> cancelTask(@PathVariable String problemID, @RequestBody CancelTaskRequest req) {
+        KubernetesClient api = makeKubernetesClient(); // TODO make the user auth connect to the database, to check
+                                                       // whether or not the user is an admin
+        api.batch().v1().jobs().inNamespace("default").withLabel("problem", problemID)
+                .withLabel("user", req.getUserId()).delete();
+        return ResponseEntity.status(HttpStatus.OK).body("Problem cancelled with ID : " + problemID);
+    }
+
+    @PostMapping("/cancel/solver/{solverName}")
+    public ResponseEntity<String> cancelSolver(@PathVariable String solverName, @RequestBody CancelSolverRequest req) {
+        KubernetesClient api = makeKubernetesClient();
+        api.batch().v1().jobs().inNamespace("default").withLabel("problem", req.getProblemID())
+                .withLabel("solver", solverName)
+                .withLabel("user", req.getUserID()).delete();
+        return ResponseEntity.status(HttpStatus.OK)
+                .body("Cancelled solver with name " + solverName + " on task" + req.getProblemID());
+    }
+
+    @PostMapping("/cancel/User")
+    public ResponseEntity<String> cancelUserTasks(@RequestBody CancelUserTasksRequest req) {
+        KubernetesClient api = makeKubernetesClient();
+        api.batch().v1().jobs().inNamespace("default").withLabel("user", req.getUserID()).delete();
+        return ResponseEntity.status(HttpStatus.OK)
+                .body("Cancelled all tasks for the user: " + req.getUserID());
+    }
+
+    private void createSolverJobs(ProblemRequest newProblem) {
+        KubernetesClient api = makeKubernetesClient();
+        for (SolverBody solver : newProblem.getSolversToUse()) {
+            List<String> command = new LinkedList<>();
+            command.add("perl");
+            command.add("-Mbignum=bpi");
+            command.add("-wle");
+            command.add("print bpi(2000)");
```

Comment:
I get that these are here so you can test it, but they should probably be changed to add the actual commands that the solver takes before pushing. Keep this stored on you local tho.

```suggestion
            command.add("python3");
            command.add("Solver.py");
            command.add(solver.getSolverName());
            command.add(solver.numberVCPU());
            command.add(newProblem.getProblemId());
            command.add(newProblem.getDataId());
```